tests/compile-and-dump/Singletons/StandaloneDeriving.hs

```diff
@@ -0,0 +1,32 @@
+module Singletons.StandaloneDeriving where
+
+import Data.Singletons.Prelude
+import Data.Singletons.Prelude.Show
+import Data.Singletons.TH
+
+$(singletons [d|
+
+  infixl 6 :*:
+  data T a = a :*: a
+  data S = S1 | S2 deriving Eq
```

Comment:
I labeled this WIP because there's one awkward point to deal with: `deriving Eq`. You'll note that I didn't standalone derive `Eq` in this example because, well, it would get ignored. That's because for some odd reason, the logic for detecting derived `Eq` instances is sequestered off in [`Data.Singletons.Single.Data`](https://github.com/goldfirere/singletons/blob/75dae57bbe06b21b3adbb82158c4767ebd695d14/src/Data/Singletons/Single/Data.hs#L59-L62), instead of `Data.Singletons.Promote.Partition`, where every other class is detected.

Is there a reason for doing it this way? If so, I need to figure out how to adapt this approach for standalone derived `Eq` instances...